ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -566,6 +569,61 @@ let definition_query server (state : State.t) uri position merlin_request =
     in
     None
 
+let workspace_symbol server (state : State.t) (params : WorkspaceSymbolParams.t)
+    =
+  let init_params = init_params state in
+  let workspace_folders = init_params.workspaceFolders in
+  let root_uri = init_params.rootUri in
+  let root_path = init_params.rootPath in
+  (* WorkspaceFolders has the most priority. Then rootUri and finally rootPath *)
+  let workspaces =
+    match (workspace_folders, root_uri, root_path) with
+    | Some (Some workspace_folders), _, _ -> workspace_folders
+    | _, Some root_uri, _ ->
+      [ Lsp.Types.WorkspaceFolder.create ~uri:root_uri
+          ~name:(Filename.basename (Uri.to_path root_uri))
+      ]
+    | _, _, Some (Some root_path) ->
+      [ Lsp.Types.WorkspaceFolder.create ~uri:(Uri.of_path root_path)
+          ~name:(Filename.basename root_path)
+      ]
+    | _ ->
+      let cwd = Sys.getcwd () in
+      [ Lsp.Types.WorkspaceFolder.create ~uri:(Uri.of_path cwd)
+          ~name:(Filename.basename cwd)
+      ]
+  in
+  let symbols_results = Workspace_symbol.run params workspaces in
+
+  let to_tuple t =
+    let fst, snd =
+      List.fold_left t
+        ~f:(fun (rs, es) x ->
+          match x with
+          | Error e -> (rs, e :: es)
+          | Ok r -> (r :: rs, es))
+        ~init:([], [])
+    in
+    (List.rev fst, List.rev snd)
```

Comment:
Does that make sense? The `Workspace_symbol.run` function returns a list of result. What I need is to get the list of `Ok symbols` on one side (to send the symbols to the client) and the list of `Error` on the other side (to send a notification to the client).

Reply:
It makes sense, but there's already a function to do this: List.partition_map